### PR TITLE
Docker Logging Driver

### LINKED ISSUE #195: Docker logging driver?
Are there plans to build a [Docker logging driver plugin](https://docs.docker.com/config/containers/logging/plugins/) based on Loki?

### PR DESCRIPTION
**Pull request by Kuqd**
This adds a Docker logging driver that can be used to send logs from Docker to Loki directly.

See the documentation for usage. For now no master or lastest plugin image have been pushed, need to merge first.

If you want to try it you without waiting:

```bash
docker plugin install grafana/loki-docker-driver:docker-driver-759556b-WIP  --alias loki --grant-all-permissions
```

Verify with `docker plugin ls`.

Then start a container: 

```bash
docker run --rm  --log-driver=loki  --log-opt loki-url="https://<user>:<password>@logs-us-west1.grafana.net/api/prom/push" grafana/grafana
```

See documentation for more. 💃 

/cc @sandlis @slim-bean @gouthamve 

Closes #195

### CODE REVIEW

**Comment by Kuqd** [comment]
would love some feedback @lfdominguez @TigerC10 @nunofernandes

File: cmd/docker-driver/driver.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+			msg.PLogMetaData.ID = buf.PartialLogMetadata.Id
+			msg.PLogMetaData.Last = buf.PartialLogMetadata.Last
+			msg.PLogMetaData.Ordinal = int(buf.PartialLogMetadata.Ordinal)
+		}
+		msg.Timestamp = time.Unix(0, buf.TimeNano)
+
+		// loki goes first as the json logger reset the message on completion.
+		if err := lf.lokil.Log(&msg); err != nil {
+			level.Error(lf.logger).Log("msg", "error pushing message to loki", "id", lf.info.ContainerID, "err", err, "message", msg)
+			continue
```
Comment (sandlis) [inline cmd/docker-driver/driver.go]: Do we want to also not write logs to the file if we fail to push them to loki?

File: cmd/docker-driver/README.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    "debug" : true,
+    "log-driver": "loki",
+    "log-opts": {
+        "loki-url": "https://<user_id>:<password>@logs-us-west1.grafana.net/api/prom/push",
+        "loki-batch-size": "400"
+    }
+}
+```
+
+> **Note**: log-opt configuration options in the daemon.json configuration file must be provided as strings. Boolean and numeric values (such as the value for max-file in the example above) must therefore be enclosed in quotes (").
```
[the remark this answers is not in the document]
Comment (Kuqd) [reply cmd/docker-driver/README.md]: Good catch

File: cmd/docker-driver/driver.go (hunk 1 above)
Comment (Kuqd) [reply cmd/docker-driver/driver.go]: yeah I think so, the driver example I used was skipping, but I think you're right.

**Pull request merged**